src/deserializer.rs

```diff
@@ -691,11 +691,7 @@ impl SerdeError for DeserializeError {
     }
 }
 
-impl StdError for DeserializeError {
-    fn description(&self) -> &str {
-        self.kind.description()
-    }
```

Comment:
This is a breaking change. Please add it back and use `#![allow(deprecated)]`.

Reply:
I incorrectly assumed std would just call the Display impl... But it can't due to the type. That's nasty.

Is it really a breaking change if the only way to hit a break is to have code like `match error.description() { "hardcoded error strings here" => do something }`? Anyone who should be doing this kind of matching would use `DeserializeErrorKind`'s variants and gain exhaustive coverage.

Reply:
The breaking change is that someone may be using `description` to output a non-empty string that describes the error. With this PR, they'd get an empty string. It's a change in behavior.

It's quite likely that in practice this wouldn't break anyone. But:

1. I don't have the tools or resources to confirm that.
2. The cost of adding `#[allow(deprecated)]` is so small (virtually zero) that it's not worth worrying about.

Reply:
Actually, this isn't where the deprecation warning comes from; but to get the strs to implement it, we'd need to call the `.description` methods on the other errors; which is deprecated.

Reply:
They don't get an empty string; they get std's own message `"description() is deprecated; use Display"`. But okay, I'll revert it.